boards/posix/nrf52_bsim/Kconfig.board

```diff
@@ -24,6 +24,8 @@ config BOARD_NRF52_BSIM
 	select HAS_HW_NRF_CLOCK
 	select HAS_HW_NRF_POWER
 	select HAS_NRFX
+	# Indicate Direction Finding Extension is supported
+	select HAS_HW_NRF_RADIO_BLE_DF
```

Comment:
This does not seem like the right thing to do.
There is no actual support for direction finding in the nrf52_bsim (52832 HW models).
It seems you want this only to be able to compile the DF unit tests which do not require the HW anyway. So maybe it should be possible to enable the needed BT_CTLR_DF_* options directly without setting this option? (@cvinayak any thoughts?)

Reply:
Correct I've done it just to run tests.
As far as I know, I have to extend nrf52_bsim to handle DF there. 
If that is correct, then I need to wanted to do in "on demand" approach. When implement some DF functionality and need to test it then add missing parts to nrf52_bsim. 
There are two questions:
- should I leave it as it is now or change board in yaml to real DK?
- is nrf52_bsim designed to be nrf52832 or can it handle other nRF SOC also?

Reply:
>As far as I know, I have to extend nrf52_bsim to handle DF there.

Maybe.. Note that the HW models won't be really able to support direction finding. You could build some empty stubbing of the registers in HW, or maybe a phony HW model which pretends to do something, but really does nothing (I have not looked into what the direction finding HW in the 833 does). But you won't be able to support simulated direction finding, as the physical layer simulation does not support it.

>should I leave it as it is now or change board in yaml to real DK?

Do you mean change the test so it runs only on a real nrf52833?

>is nrf52_bsim designed to be nrf52832 or can it handle other nRF SOC also?

The nrf52_bsim uses the nrf52 HW models, which as today target a 52832 (I needed to chose something whenever there was differences, so I targeted that one). I actually haven't looked in detail into what is different in a 52833.
A relevant thing: the nrf52_bsim does not have any configuration to select modeling one or another IC. If you guys would want to start having several SOCs options in the nrf52_bsim we'd need to add some config options to select which.

Reply:
> Maybe.. Note that the HW models won't be really able to support direction finding. You could build some empty stubbing of the registers in HW, or maybe a phony HW model which pretends to do something, but really does nothing (I have not looked into what the direction finding HW in the 833 does). But you won't be able to support simulated direction finding, as the physical layer simulation does not support it.

Yes, I know that. In my humble opinion, simulation of doing something on air is enough to implement tests. To validate code, there is a need to have expected HW behavior only. So phony stubs should be enough.

> Do you mean change the test so it runs only on a real nrf52833?

Yes, that is what I meant. 

> The nrf52_bsim uses the nrf52 HW models, which as today target a 52832 (I needed to chose something whenever there was differences, so I targeted that one). I actually haven't looked in detail into what is different in a 52833.
A relevant thing: the nrf52_bsim does not have any configuration to select modeling one or another IC. If you guys would want to start having several SOCs options in the nrf52_bsim we'd need to add some config options to select which.

The thing is, DF is not supported by all nRF SoC. To make it possible to tests controller for DF, the missing implementation in nrf52_board should be allowed. Or we have to test controller with use  of real hardware. 
I'm not sure if approach to force build system to enable some DF Kconfigs is desired. 
What do you thing guys?

Reply:
> It seems you want this only to be able to compile the DF unit tests which do not require the HW anyway. So maybe it should be possible to enable the needed BT_CTLR_DF_* options directly without setting this option? 

As the real h/w will not be used in simulation, introduce HAL interface for logical functionalities (which in turn use the nRFx interfaces on DK). For the unit tests, these stubs be defined in the tests.

> should I leave it as it is now or change board in yaml to real DK?
No, unit tests be in nrf52_bsim, and stubs handle the test case requirements.

> is nrf52_bsim designed to be nrf52832 or can it handle other nRF SOC also?
Currently nrf52_bsim is restricted to nRF52832 feature set. If there is requirement in the future, we can migrate it to support latest new h/w features. But we will only keep single simulation target (it will be expensive to maintain multiple simulation targets).